.gitignore

```diff
@@ -1,4 +1,81 @@
-.idea
-.history
+# Miscellaneous
+*.lock
+*.log
+*.pyc
+*.swp
 .DS_Store
+.atom/
+.buildlog/
+.history
+.svn/
+
+# IntelliJ related
 *.iml
+*.ipr
+*.iws
+.idea/
+
+# Visual Studio Code related
+.vscode/
+
+# Flutter repo-specific
+/bin/cache/
+/dev/benchmarks/mega_gallery/
+/dev/bots/.recipe_deps
+/dev/bots/android_tools/
+/dev/docs/doc/
+/dev/docs/lib/
+/packages/flutter/coverage/
+
+# Flutter/Dart/Pub related
+**/doc/api/
+.flutter-plugins
+.packages
+.pub/
+build/
+flutter_*.png
+linked_*.ds
+pubspec.yaml
```

Comment:
are we sure we don't want this?

Reply:
LOL. Yeah, I guess so.  This came from the dev/docs gitignore, so I just limited it to that one.